### PR TITLE
issues 10, 20, 25, 26

### LINKED ISSUE #10: mixed content
E.g. json content in a CSV file > switch reference formulation on field level. 
Concrete examples needed + declarative description to be elaborated.

### LINKED ISSUE #20: index in case of (nested) csv
How to calculate the index in case of csv? 
Until now I supposed that we keep the index of the row
| # | name.# | name   | birthyear.# | birthyear |
| - | ------ | ------ | ----------- | --------- |
| 0 | 0      | alice  | 0           | 1995      |
| 1 | 1      | bob    | 1           | 1999      |
| 2 | 2      | tobias | 2           | 2005      |

However, is this in line with how we calculated the index for json file? There the index is 0 if there is only on value for the logical iteration. In case of csv there standard only one value for the logical iteration, so then the index of all fields should be 0, except for the overall index #. 

| # | name.# | name   | birthyear.# | birthyear |
| - | ------ | ------ | ----------- | --------- |
| 0 | 0      | alice  | 0           | 1995      |
| 1 | 0    | bob    | 0           | 1999      |
| 2 | 0      | tobias | 0          | 2005      |

And what to do with a nested csv column? In that case it makes more sense to maintain the index of the row, otherwise we cannot reconstruct the order. 
```
{
  "people": [
    {
      "name": "alice",
      "hobbies": "id;type\n1;volleybal\n2;basketball\n3;horses"
    },
    {
      "name": "bob",
      "hobbies": "id;type\n1;football"
    }
  ]
}
```
| # | name  | name.# | hobbies | hobbies.# | hobbies.id | hobbies.id.# | hobbies.type | hobbies.type.# |
| - | ----- | ------ | ------- | --------- | ---------- | ------------ | ------------ | -------------- |
| 0 | alice | 0      | (...)   | 0         | 1          | 0            | volleybal    | 0              |
| 0 | alice | 0      | (...)   | 0         | 2          | 1            | basketball   | 1              |
| 0 | alice | 0      | (...)   | 0         | 3          | 2            | horses       | 2              |
| 1 | bob   | 0      | (...)   | 0         | 1          | 0            | football     | 0              |

Should we define the algorithm to calculate the index for a field per reference formulation?

### LINKED ISSUE #25: Example 9 and Usefulness of nested fields.
Example 9 is not complete, as the weight is not rendered as a field. Furthermore, is it necessary to create a field for "Item"? We think it just adds confusion. Related to this: is nesting in fields really necessary?

### LINKED ISSUE #26: A Field is an ExpressionMap
Hi, spec says "A field is an [ExpressionMap](http://w3id.org/rml/core/spec#dfn-expression-map) that gives a name to an [expression](https://w3id.org/rml/core/spec#dfn-expressions)."

Actually, also in the ontology, Fields are subclasses of expression maps. Thus, I would phrase this more explicitly as: "A field is a particular kind of `ExpressionMap`".

As for the `ExpressionMap` specification, we also suggest to list the additional properties of Fields (e.g., `rml:fieldName` or `rml:field`), instead of having them informally introduced in the text only.

### PR DESCRIPTION
**Pull request by elsdvlee**
## What's changed

Added mixed content specification. 
Adapted field specification. 
Adapted example 9.

## Change checklist
- [x] updated ontology, where necessary
- [x] updated shapes, where necessary
- [x] added or updated test cases, where necessary
- [x] any TODOs have been turned into trackable issues and referenced where necessary

## Issue reference

Fixes #10.
Fixes #20.
Fixes #25.
Fixes #26.

### CODE REVIEW

**Review by pmaria — CHANGES_REQUESTED** [review]
Some minor points in comments.

As discussed I have some other concerns regarding aspects that are currently described as part of an `rml:Source` like `rml:null` or `CSVW` descriptions which we cannot reuse in fields in the current state of rml-lv vs rml-io. But we can raise a different issue for this.

File: spec/section/fields.md [hunk 1]
```diff
@@ -2,25 +2,23 @@
...
+| `rml:field`                  | `rml:LogicalView`,`rml:Field` | `rml:Field`                |
 
-### Field parents {#fieldparents}
+### Field parents {#field-parents}
 
-A [=field=] MUST have a <dfn data-lt="field-parent">parent</dfn> that is either a <!-- TODO reference to core, dependent on https://github.com/kg-construct/rml-core/issues/127-->[logical source]() or another [=field=]. The parent relation MUST not contain cycles: it is tree-shaped with a logical source as its root. The transitive parents of a [=field=], i.e., the [=field=]'s parent, the parent of the [=field=]'s parent, etcetera, are fittingly called the [=field=]'s <dfn>ancestors</dfn>. 
+A [=field=] MUST have a <dfn data-lt="field parent">parent</dfn> that is either a <!-- TODO reference to core, dependent on https://github.com/kg-construct/rml-core/issues/127-->[logical source]() or another [=field=]. The parent relation MUST not contain cycles: it is tree-shaped with a logical source as its root. The transitive parents of a [=field=], i.e., the [=field=]'s parent, the parent of the [=field=]'s parent, etcetera, are fittingly called the [=field=]'s <dfn>ancestors</dfn>.
```
[the remark this answers is not in the document]
Comment (elsdvlee) [reply spec/section/fields.md:19]: It is the Logical Source of the Logical View that has the correct reference formulation.

[the remark this answers is not in the document]
Comment (pmaria) [reply spec/section/fields.md:19]: ok

File: spec/section/fields.md [hunk 2]
```diff
@@ -240,17 +238,18 @@ Note some columns in the table below have been shortened for brevity.
...
 
-A [=field=] MUST have an a <!-- TODO reference to core, dependent on https://github.com/kg-construct/rml-core/issues/127-->reference formulation. If no <!-- TODO reference to core, dependent on https://github.com/kg-construct/rml-core/issues/127-->reference formulation is declared for the [=field=], the [=field=] has the same <!-- TODO reference to core, dependent on https://github.com/kg-construct/rml-core/issues/127-->reference formulation as its parent.
-If a field has a declared reference formulation, it MUST also have an iterator. By default, the iterator is considered a row, if not specified. 
+A [=field=] MUST have an a <!-- TODO reference to core reference formulation when available-->[reference formulation]() and an <!-- TODO reference to core iterator when available-->[iterator]() . 
+If no reference formulation is declared for a field, the reference formulation of the [=field parent=] is implied.
+If no iterator is declared for a field, a default iterator is implied. If the reference formulation of the field is `rml:JSONPath` or `rml:XMLPath`, the default iterator is the root of the document. Otherwise, the default iterator is a row.
```
[the remark this answers is not in the document]
Comment (elsdvlee) [reply spec/section/fields.md:245]:
```suggestion
If no iterator is declared for a field, the default iterator of the field's reference formulation is implied.
```

[the remark this answers is not in the document]
Comment (elsdvlee) [reply spec/section/fields.md:245]: I will make an issue in rml-io.

[the remark this answers is not in the document]
Comment (elsdvlee) [reply spec/section/fields.md:245]: https://github.com/kg-construct/rml-io/issues/87

[the remark this answers is not in the document]
Comment (pmaria) [reply spec/section/fields.md:245]: lgtm

**Review by pmaria — APPROVED** [review]

**Comment by elsdvlee** [comment]
Made [issue 34](https://github.com/kg-construct/rml-lv/issues/34) for the open unresolved point.

**Pull request merged**